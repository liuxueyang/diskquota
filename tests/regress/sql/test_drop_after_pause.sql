CREATE DATABASE test_drop_after_pause;

\c test_drop_after_pause

CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();

SELECT diskquota.enable_hardlimit();

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO SX.a SELECT generate_series(1,1000000); -- expect insert fail

SELECT diskquota.disable_hardlimit();
SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

\c contrib_regression

DROP DATABASE test_drop_after_pause;
