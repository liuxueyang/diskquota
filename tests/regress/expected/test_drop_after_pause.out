CREATE DATABASE test_drop_after_pause;
\c test_drop_after_pause
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
CREATE EXTENSION diskquota;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

SELECT diskquota.enable_hardlimit();
 enable_hardlimit 
------------------
 
(1 row)

CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO SX.a SELECT generate_series(1,1000000); -- expect insert fail
ERROR:  schema's disk space quota exceeded with name:16933  (seg2 127.0.0.1:6004 pid=24622)
SELECT diskquota.disable_hardlimit();
 disable_hardlimit 
-------------------
 
(1 row)

SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c contrib_regression
DROP DATABASE test_drop_after_pause;
